pkg/client/unversioned/clientcmd/api/v1/types.go

```diff
@@ -54,7 +54,10 @@ type Preferences struct {
 type Cluster struct {
 	// Server is the address of the kubernetes cluster (https://hostname:port).
 	Server string `json:"server"`
-	// APIVersion is the preferred api version for communicating with the kubernetes cluster (v1, v2, etc).
+	// DEPRECATED: APIVersion is the preferred api version for communicating with the kubernetes cluster (v1, v2, etc).
+	//  Because a cluster can run multiple API groups and potentially multiple versions of each, it no longer makes sense to specify
+	//  a single value for the cluster version. This field isn't really needed anyway, so we're deprecating it without a replacement.
+	//  It will be ignored if it is present.
```

Comment:
> I like this idea in general, but I'm not sure of "must" versus "must if available". I don't think we have to debate it before this changes in, but we currently have "must if available" and I've been pretty pleased with its operation.

I also think we should "must if avalaible", and let NegotiateVersion report "fall back to XXX version" if the preferred version is not available.

> What happens in this scenario if you have "experimental/v1" and "experimental/v2"?
Will we need to express two preferred versions for one group?